packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -208,29 +232,80 @@ export const render = async (req, res, next) => {
     }
 }
 
-const renderAppHtml = (req, res, error, appData) => {
-    const {App, appState, routes, routerContext, location, extractor, deviceType} = appData
-
-    let appJSX = (
-        <Router location={location} context={routerContext}>
-            <DeviceContext.Provider value={{type: deviceType}}>
-                <AppConfig locals={res.locals}>
-                    <Switch error={error} appState={appState} routes={routes} App={App} />
-                </AppConfig>
-            </DeviceContext.Provider>
-        </Router>
+const getAppJSX = (req, res, error, appData) => {
+    const {App, appState = {}, deviceType, location, queryClient, routerContext, routes} = appData
+
+    return (
+        <ExpressContext.Provider value={{req, res}}>
+            <QueryClientProvider client={queryClient}>
```

Comment:
```suggestion
            <QueryClientProvider client={queryClient} contextSharing={true}>
```

In response to Kevin's question on Slack recently: I think both `pwa-kit-react-sdk` and `commerce-sdk-react` packages can mount their own QueryClientProvider, that is, if `commerce-sdk-react` is meant to be a standalone library. By setting `contextSharing` to true, it looks like it's ok to have multiple QueryClientProvider in the component tree, and only one instance will be used. 

What `contextSharing` option does: https://react-query-v3.tanstack.com/reference/QueryClientProvider

Reply:
`commerce-sdk-react` isn't going to be mounting it's own provider. That being said, it doesn't mean that someone wouldn't add their own for some weird reason. The question is, do we want to make that decision for the user.. also does context only get shared amongst providers that have that value set to true? hmm